src/Writers/HTMLWriter.jl

```diff
@@ -4,24 +4,471 @@ Provides the [`render`](@ref) methods to write the documentation as HTML files
+    scripts :: Vector{Compat.String}
+    documenter_js :: Compat.String
+    documenter_css :: Compat.String
+    search_index :: IOBuffer
+    search_index_js :: Compat.String
+end
+HTMLContext(doc) = HTMLContext(doc, "", [], "", "", IOBuffer(), "")
+
+"""
+Returns a page from the [`HTMLContext`](@ref).
+"""
+getpage(ctx, path) = ctx.doc.internal.pages[path]
+getpage(ctx, navnode::Documents.NavNode) = getpage(ctx, get(navnode.page))
 
-# TODO
 
 function render(::Writer{Formats.HTML}, doc::Documents.Document)
-    error("HTML rendering is unsupported.")
+    !isempty(doc.user.sitename) || error("HTML output requires `sitename`.")
+
+    ctx = HTMLContext(doc)
+    ctx.search_index_js = "search_index.js"
+
+    copy_asset("layout.css", doc)
+    copy_asset("style.css", doc)
+    ctx.documenter_css = copy_asset("documenter.css", doc)
+
+    local_assetsdir = joinpath(pwd(), "assets")
+    let src = joinpath(local_assetsdir, "logo.png")
+        if isfile(src)
+            dst = joinpath(doc.user.build, "logo.png")
+            cp(src, dst, remove_destination=true)
+            ctx.logo = "logo.png"
+        end
+    end
+
+    ctx.documenter_js = copy_asset("documenter.js", doc)
+
+    for navnode in doc.internal.navlist
+        render_page(ctx, navnode)
+    end
+
+    open(joinpath(doc.user.build, ctx.search_index_js), "w") do io
+        println(io, "var documenterSearchIndex = {\"docs\": [\n")
+        write(io, takebuf_string(ctx.search_index))
+        println(io, "]}")
+    end
+end
+
+"""
+Copies an asset from Documenters `assets/html/` directory to `doc.user.build`.
+Returns the path of the copied asset relative to `.build`.
+"""
+function copy_asset(file, doc)
+    src = joinpath(Utilities.assetsdir(), "html", file)
+    dst = joinpath(doc.user.build, "assets", file)
+    isfile(src) || error("Asset '$file' not found at $(abspath(src))")
+    ispath(dirname(dst)) || mkpath(dirname(dst))
+    ispath(dst) && Utilities.warn("Overwriting '$dst'.")
+    cp(src, dst, remove_destination=true)
+    normpath(joinpath("assets", file))
+end
+
+
+# Page
+# ------------------------------------------------------------------------------
+
+"""
+Constructs and writes the page referred to by the `navnode` to `.build`.
+"""
+function render_page(ctx, navnode)
+    @tags html body
+
+    page = getpage(ctx, navnode)
+
+    head = render_head(ctx, navnode)
+    navmenu = render_navmenu(ctx, navnode)
+    article = render_article(ctx, navnode)
+
+    htmldoc = DOM.HTMLDocument(
+        html[:lang=>"en"](
+            head,
+            body(navmenu, article)
+        )
+    )
+    open(Formats.extension(Formats.HTML, page.build), "w") do io
+        print(io, htmldoc)
+    end
+end
+
+function render_head(ctx, navnode)
+    @tags head meta link script title
+    src = get(navnode.page)
+    page_title = "$(mdflatten(pagetitle(ctx, navnode))) · $(ctx.doc.user.sitename) documentation"
+    head(
+        meta[:charset=>"UTF-8"](),
```

Comment:
We *could* add a 0-arg method that just passes the `Node` back to avoid additional allocations. I find the extra `()` redundant, but perhaps that just from knowing how it all works underneath.